pkg/operator/controller/extension/extension/runtime/runtime.go

```diff
@@ -84,52 +90,106 @@ func (d *deployer) createOrUpdateResources(ctx context.Context, extension *opera
+	if err := managedresources.CreateForSeed(ctx, d.runtimeClientSet.Client(), namespace, mrName, false, renderedChart.AsSecretData()); err != nil {
 		return fmt.Errorf("failed creating ManagedResource: %w", err)
 	}
 
-	if err := managedresources.WaitUntilHealthyAndNotProgressing(ctx, d.runtimeClientSet.Client(), d.gardenNamespace, mrName); err != nil {
+	if err := managedresources.WaitUntilHealthyAndNotProgressing(ctx, d.runtimeClientSet.Client(), namespace, mrName); err != nil {
 		return fmt.Errorf("failed waiting for ManagedResource to be healthy: %w", err)
 	}
 	return nil
 }
 
 func (d *deployer) deleteResources(ctx context.Context, log logr.Logger, extension *operatorv1alpha1.Extension) error {
 	mrName := managedResourceName(extension)
+	namespace := namespaceName(extension)
 
-	log.Info("Deleting extension ManagedResource for runtime cluster if present", "managedResource", client.ObjectKey{Name: mrName, Namespace: d.gardenNamespace})
-	if err := managedresources.DeleteForSeed(ctx, d.runtimeClientSet.Client(), d.gardenNamespace, mrName); err != nil {
+	log.Info("Deleting extension ManagedResource for runtime cluster if present", "managedResource", client.ObjectKey{Name: mrName, Namespace: namespace})
+	if err := managedresources.DeleteForSeed(ctx, d.runtimeClientSet.Client(), namespace, mrName); err != nil {
 		return fmt.Errorf("failed deleting ManagedResource: %w", err)
 	}
 
-	if err := managedresources.WaitUntilDeleted(ctx, d.runtimeClientSet.Client(), d.gardenNamespace, mrName); err != nil {
+	if err := managedresources.WaitUntilDeleted(ctx, d.runtimeClientSet.Client(), namespace, mrName); err != nil {
 		return fmt.Errorf("failed waiting for ManagedResource to be deleted: %w", err)
 	}
+
+	if err := d.deleteNamespace(ctx, namespace); err != nil {
+		return fmt.Errorf("failed deleting namespace %q: %w", namespace, err)
+	}
+
+	return nil
+}
+
+func (d *deployer) deleteExtensionManagedResourceInGardenNamespace(ctx context.Context, log logr.Logger, extension *operatorv1alpha1.Extension) error {
+	mrName := managedResourceName(extension)
+	namespace := v1beta1constants.GardenNamespace
+
+	log.Info("Deleting old extension ManagedResource for runtime cluster if present", "managedResource", client.ObjectKey{Name: mrName, Namespace: namespace})
+	if err := managedresources.DeleteForSeed(ctx, d.runtimeClientSet.Client(), namespace, mrName); err != nil {
+		return fmt.Errorf("failed deleting old ManagedResource: %w", err)
+	}
+
+	if err := managedresources.WaitUntilDeleted(ctx, d.runtimeClientSet.Client(), namespace, mrName); err != nil {
+		return fmt.Errorf("failed waiting for old ManagedResource to be deleted: %w", err)
+	}
+
+	return nil
+}
+
+func (d *deployer) ensureNamespace(ctx context.Context, name string) error {
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
```

Comment:
We need the `gardener.cloud/role: extension` label here to benefit from automatic network policy management.

Should we add `high-availability-config.resources.gardener.cloud/consider` and `high-availability-config.resources.gardener.cloud/zones`, too?

Reply:
Done!
@timuthy Can you validate if labels and annotations are correct?

Reply:
Looks good 👍